### ISSUE TITLE
Monitoring system tests:  Collision on teardown / group.delete

### ISSUE THREAD

**Issue by tseaver** [GoogleCloudPlatform/gcloud-python#2092]
From: https://travis-ci.org/GoogleCloudPlatform/gcloud-python/builds/151551907#L1069-L1079

```python
======================================================================
ERROR: test_create_group (monitoring.TestMonitoringGroups)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/GoogleCloudPlatform/gcloud-python/system_tests/monitoring.py", line 190, in tearDown
    group.delete()
  File "/home/travis/build/GoogleCloudPlatform/gcloud-python/.tox/system-tests3/lib/python3.4/site-packages/gcloud/monitoring/group.py", line 227, in delete
    self.client.connection.api_request(method='DELETE', path=self.path)
  File "/home/travis/build/GoogleCloudPlatform/gcloud-python/.tox/system-tests3/lib/python3.4/site-packages/gcloud/connection.py", line 347, in api_request
    error_info=method + ' ' + url)
gcloud.exceptions.ServiceUnavailable: 503 Write collision, please retry. (DELETE https://monitoring.googleapis.com/v3/projects/precise-truck-742/groups/5292676180999102311)
```

**Comment by rimey** [comment]
"503 Write collision, please retry" is slightly surprising here. We'll look into it, but it would be good to reproduce this in isolation, and also determine if it is deterministic. Tomorrow.

**Comment by dhermes** [comment]
Thanks. We were surprised too; at the least we'd expect a well-formed error "collision" to come with a "409 Conflict" status code. It is not deterministic; at least the determining factors aren't ones we know of, since it doesn't fail on every test run.

**Comment by rimey** [comment]
By my reading, a 409 would not be appropriate for an error that says "please retry."

**Comment by dhermes** [comment]
Well a 503 Service Unavailable isn't either. I'm not sure why the suggestion of retry overrides the rough equivalence "collision" == "conflict". What 4xx code would you find more appropriate?

**Comment by rimey** [comment]
503 Service Unavailable really is the code we are supposed to return for a failed request that can be retried as-is. Give us time to investigate. Then we can consider whether the error code and message text are optimal.

**Comment by dhermes** [comment]
> Give us time to investigate. Then we can consider

SGTM

**Comment by dhermes** [comment]
Closed by #2095
